TheMovieDB/TheMovieDB/Network/NetworkManager.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  NetworkManager.swift
+//  TheMovieDB
+//
+//  Created by Juan Martin Rezk Elso on 1/9/21.
+//
+
+import Foundation
+
+protocol Manager {
+    func retrieveMovies(completion: @escaping (Result<intermediaryJson?, Error>) -> Void)
+}
+
+enum NetworkError: Error {
+    case urlError
+    case serverError
+    case parsingError
+}
+
+class NetworkManager: Manager {
+    func retrieveMovies(completion: @escaping (Result<intermediaryJson?, Error>) -> Void) {
+        guard let url = URL(string: "https://api.themoviedb.org/3/movie/top_rated?api_key=d0b82d9ad21c5aa4e12d4cf1acfb1155&language=en-US&page=1") else {
```

Comment:
you shouldn't have the entire url here in a method because you could have multiple paths to call, extract it as a private property